didcomm/did_doc/did_doc.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import List
+
+from didcomm.common.types import DID_URL, DID
+
+
+class DIDDoc(ABC):
+    """DID DOC abstraction (https://www.w3.org/TR/did-core/#dfn-did-documents)"""
+
+    @abstractmethod
```

Comment:
It feels more natural to access these values as `diddoc.did` as opposed to `diddoc.did()`. My vote is for most of these abstract methods to be abstract properties.

Reply:
Fixed, thank you.